README.md

```diff
@@ -132,32 +128,110 @@ and some optional params can be provided as well:
+- In Preprocessing tab: add JSONPath step with parameter: `$.10`
+![image](https://user-images.githubusercontent.com/14870891/45901353-507af600-bdeb-11e8-9350-c7a0c5ab8e7f.png)
+![image](https://user-images.githubusercontent.com/14870891/45901388-67b9e380-bdeb-11e8-850c-e95c3bb4aa7e.png)
+
+
+
+And the second item:  
+- Key: register11    
+- Type: Dependent item
+- Master item: Modbus bulk request
+- Type of information: Numeric(Unsigned)  
+- In Preprocessing tab: add JSONPath step with parameter: `$.11`
+
+Congratulations! You've just collected two metrics with the single Modbus command.  
+Note: You can find examples how mass data collection works in Zabbix [here](https://blog.zabbix.com/zabbix-3-4-mass-data-collection-using-mercury-and-smartmontools-as-an-example/5784/).
+
+### 5.2. Another Example
+But datatype expressions can be more advanced. First, you can combine together different datatypes -   
+`uint16+double+float`, and you can define how many times each datatype should be repeated - 
+`2*uint16+double+float*3`. So in a key `modbus_read[{$MODBUS_PORT},{$MODBUS_SLAVE},10,3,2*uint16+double+float*3]` you would get a JSON:
+```
+{
+  "10":123,
+  "11":321,
+  "12":0.12345,
+  "16":100.10,
+  "18":100.10,
+  "20":100.10
+}  
+```
+Note two things here. First, since `double` is 4 words long (64bit/16bit = 4), next request register address is actually `16`. Same applies to the following registers: since `float` datatype is 2 words long (32bit/16bit = 2) you would see next keys as `18` and `20`.
+
+### 5.3. Example with skip
+Modbus protocol supports only sequential read of registers. But what if you don't need all of them? For that, there is a special keyword `skip` you can use in order to retrieve multiple registers that are not located together. Example: Let's retrieve two `int16` from registers with addresses of `10` and `20`. Zabbix key 
+`modbus_read[{$MODBUS_PORT},{$MODBUS_SLAVE},10,3,int16+10*skip+int16]` would give you:
+```
+{
+  "10":123,
+  "20":-123
+}
+```
+As you may noticed, `skip` is one word (16bit) long.  
+
+
     
-## 4. Troubleshooting and Testing Modbus
+## 6. Troubleshooting and Testing Modbus
 
-Testing Modbus connectivity is easy with *modpoll* command utility.  
-You may try to grab some modbus registers with it before you try to do it with Zabbix.  
-http://www.modbusdriver.com/modpoll.html
+Testing Modbus connectivity is easy with [*modpoll*](http://www.modbusdriver.com/modpoll.html) command utility.  
+You may try to grab some Modbus registers with it before you try to do it with Zabbix.  
 
-If using libzbxmodbus with Zabbix agent, you can also save time and test responses first with `zabbix_get`, for example:
+
+If you use libzbxmodbus with Zabbix agent, then you can also save time and test responses first with `zabbix_get`, for example:
 `zabbix_get -s localhost -kmodbus_read["/dev/ttyS1 9600 N",9,0x1518,3,l,1,0]`
   
-## 5. Known Issues and Limitations  
+## 7. Known Issues and Limitations  
```

Comment:
It seems there is a [way](https://support.typora.io/Auto-Numbering/) to make headings auto-numbered.